playwright-dotnet/PlaywrightParallelTest.cs

```diff
@@ -0,0 +1,122 @@
+                    browserstackOptions.Add("name", test_name);
+                    browserstackOptions.Add("os", os);
+                    browserstackOptions.Add("os_version", os_version);
+                    browserstackOptions.Add("browser", browser_name);
+                    browserstackOptions.Add("browser_version", browser_version);
+                    browserstackOptions.Add("browserstack.username", "BROWSERSTACK_USERNAME");
+                    browserstackOptions.Add("browserstack.accessKey", "BROWSERSTACK_ACCESS_KEY");
+                    capsJson = JsonConvert.SerializeObject(browserstackOptions);
+                    task = Executetestwithcaps(capsJson);
+                    await task;
+                    break;
+                default:
+                    browserstackOptions.Add("build", build_name);
+                    browserstackOptions.Add("name", test_name);
+                    browserstackOptions.Add("os", os);
+                    browserstackOptions.Add("os_version", os_version);
+                    browserstackOptions.Add("browser", browser_name);
+                    browserstackOptions.Add("browser_version", browser_version);
+                    browserstackOptions.Add("browserstack.username", "BROWSERSTACK_USERNAME");
+                    browserstackOptions.Add("browserstack.accessKey", "BROWSERSTACK_ACCESS_KEY");
+                    capsJson = JsonConvert.SerializeObject(browserstackOptions);
+                    task = Executetestwithcaps(capsJson);
+                    await task;
+                    break;
+            }
+        } catch (Exception e) {
+            Console.WriteLine(e);
+        }
+    }
+
+    //Executetestwithcaps function takes capabilities from 'SampleTestCase' function and executes the test
+    public static async Task Executetestwithcaps(string capabilities)
+    {
+        using var playwright = await Playwright.CreateAsync();
+        string cdpUrl = "wss://cdp.browserstack.com/playwright?caps=" + Uri.EscapeDataString(capabilities);
+
+        await using var browser = await playwright.Chromium.ConnectAsync(cdpUrl);
+        var page = await browser.NewPageAsync();
+        try {
+            await page.GotoAsync("https://www.google.co.in/");
+            await page.Locator("[aria-label='Search']").ClickAsync();
+            await page.FillAsync("[aria-label='Search']", "BrowserStack");
+            await page.Locator("[aria-label='Google Search'] >> nth=0").ClickAsync();
+            var title = await page.TitleAsync();
+
+            if (title == "BrowserStack - Google Search")
+            {
+                // following line of code is responsible for marking the status of the test on BrowserStack as 'passed'. You can use this code in your after hook after each test
+                await page.EvaluateAsync("_ => {}", "browserstack_executor: {\"action\":\"setSessionStatus\",\"arguments\":{\"status\":\"passed\",\"reason\":\"Title matched\"}}");
+            }
```

Comment:
Else Title didnot match, catch the error and in line number 116 mark the reason with error string